example_notebooks/user_stories/user_story_2/user_story_2_best_case_researcher.py

```diff
@@ -0,0 +1,121 @@
+"""
+User story 2 (best case) as researcher
+"""
+
+import logging
+import os
+import pickle
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder
+
+from aisdc.attacks.target import Target  # pylint: disable=import-error
+from aisdc.safemodel.classifiers import (
+    SafeDecisionTreeClassifier,  # pylint: disable=import-error
+)
+
+
+def main():
+    """
+    Create and train a model to be released
+    """
+    directory = "training_artefacts/"
+    print("Creating directory for training artefacts")
+
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    print()
+    print("Acting as researcher...")
+    print()
+
+    filename = "user_stories_resources/dataset_26_nursery.csv"
+    print("Reading data from " + filename)
+    data = pd.read_csv(filename)
+
+    print()
+
+    y = np.asarray(data["class"])
+    x = np.asarray(data.drop(columns=["class"], inplace=False))
+
+    n_features = np.shape(x)[1]
+    indices: list[list[int]] = [
+        [0, 1, 2],  # parents
+        [3, 4, 5, 6, 7],  # has_nurs
+        [8, 9, 10, 11],  # form
+        [12, 13, 14, 15],  # children
+        [16, 17, 18],  # housing
+        [19, 20],  # finance
+        [21, 22, 23],  # social
+        [24, 25, 26],  # health
+    ]
+
+    row_indices = np.arange(np.shape(x)[0])
+
+    # [Researcher] Split into training and test sets
+    # target model train / test split - these are strings
+    (
+        x_train_orig,
+        x_test_orig,
+        y_train_orig,
+        y_test_orig,
+        indices_train,
+        indices_test,
+    ) = train_test_split(
+        x,
+        y,
+        row_indices,
+        test_size=0.5,
+        stratify=y,
+        shuffle=True,
+    )
+
+    # Preprocess dataset
+    # one-hot encoding of features and integer encoding of labels
+    label_enc = LabelEncoder()
+    feature_enc = OneHotEncoder()
+    x_train = feature_enc.fit_transform(x_train_orig).toarray()
+    y_train = label_enc.fit_transform(y_train_orig)
+    x_test = feature_enc.transform(x_test_orig).toarray()
+    y_test = label_enc.transform(y_test_orig)
+
+    logging.getLogger("attack-reps").setLevel(logging.WARNING)
+    logging.getLogger("prep-attack-data").setLevel(logging.WARNING)
+    logging.getLogger("attack-from-preds").setLevel(logging.WARNING)
+
+    # Build a model
+    model = SafeDecisionTreeClassifier(random_state=1)
+    model.fit(x_train, y_train)
+    msg, disclosive = model.preliminary_check()
+
```

Comment:
line 92 I think we should illustrate them calling model.request_release() but not giving it the target object.
In that case it will still produce the report on whether they did things like over-ride TRE-preferred values for hyper-parameters, change the models' apparent values for hyper-parameters  after calling the fit function, or change the actual content of the model (neural net, forest or tree) after calling fit (I.e. hide data in it)

Reply:
continued: although I'm not sure that your generate report code would pick that up? If not needs adding as an issue for the next sprint